esp-hal/src/uart.rs

```diff
@@ -511,6 +512,8 @@ impl Default for Config {
     }
 }
 
+#[derive(procmacros::BuilderLite)]
+#[non_exhaustive]
```

Comment:
At the same time we can mark this as `#[derive(Clone, Copy, PartialEq, Eq, Hash, Debug)]` + defmt::Format as well, I think.